gui/lib/data/models/popup_metadata.dart

```diff
@@ -29,6 +29,8 @@ final class DecisionPopupMetadata extends PopupMetadata {
   final String noButtonText;
   final String yesButtonText;
   final Function(WidgetRef ref) yesAction;
+  final bool autoClose;
+  final String? navigateToRoute;
```

Comment:
Maybe this can be `AppRoute` enum? Slightly better type safety. 
Then in `_yesButton`, you can call our extension on the context `context.navigateTo(AppRoute route)` instead `context.go(String route)`